### ISSUE TITLE
IntelliJ crash with JCEF preview

### ISSUE THREAD

**Issue by cvmocanu** [asciidoctor/asciidoctor-intellij-plugin#593]
## Observed vs. expected behavior
When opening the following asciidoc file, the JCEF preview browser causes IntelliJ to crash (without any crash log).
The problem is not reproducible with the Swing preview browser.
```asciidoc
= Tools for the terminal

[[format-json-logs]]
== Format JSON logs

[source,shell script]
----
#
# Description : makes JSON-per-line logs readable again
# Requirements: cat, tee, grep, jq (for JSON processing)
# JSON format: this function will use the following fields from JSON: "@timestamp", "level", "logger_name", "message", and optionally: "X-B3-TraceId", "X-B3-SpanId", and "X-B3-ParentSpanId"
#
# To use it, just pipe a JSON into this function, e.g. "kubectl logs {pod-name} | format_json_logs"
#
format_json_logs() {
    cat | tee >(grep -v "^{") | grep "^{" | jq -r 'def lpad(string;len;fill): if len == 0 then string else (fill * (len  - (string|length)))[0:len] + string end; def rpad(string;len;fill): if len == 1 then string else string + (fill * (len  - (string|length)))[0:len] end; ("»»»»»»»»»» " + .["@timestamp"] + " " + rpad(.level;5;" ") + " [" + .logger_name + "," + (.["X-B3-TraceId"] // "/") + "-" + (.["X-B3-SpanId"] // "/") + "-" + (.["X-B3-ParentSpanId"] // "/") + "]" + " " + .message)'
}
----

== Download formatted kubernetes logs

Note that this function need the function described in <<format-json-logs>>.

[source, shell script]
----
#
# Name         : kubernetes logs
# Description  : download the logs of one or multiple pods, transforms the JSON into a readable form, and saves them to files.
# Requirements : format_json_logs, kubectl, fzf (for interactive selection)
#
# How to use
# ----------
# Call "kl". This will open an interactive list of pods.
# To cancel at any time, press Esc.
# To get the logs from a single pod, just press Enter.
# To get the logs from multiple pods, use the arrows to move the cursor, and Tab to toggle the selection. Press Enter when done.
# Note: you can use normal readline keyboard shortcuts, like Ctrl+W to delete a word backward, or Alt+D to delete a word forward.
#
kl() {
    local pod_names=$(kubectl get pods | fzf -m | cut -d' ' -f 1)

    if [[ -z "$pod_names" ]]; then
        return
    fi

    while IFS= read -r pod_name ; do
        echo "getting logs for pod [$pod_name]"
        kubectl logs "$pod_name" | format_json_logs > "$pod_name.log"
    done <<< "$pod_names"
}
----
```
## Environment

AsciiDoc plugin version: ``0.31.38``

IntelliJ version:
```
IntelliJ IDEA 2020.3 EAP (Ultimate Edition)
Build #IU-203.5251.39, built on October 23, 2020
IntelliJ IDEA EAP User
Expiration date: November 22, 2020
Runtime version: 11.0.8+10-b1145.3 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.4.0-52-generic
GC: ParNew, ConcurrentMarkSweep
Memory: 1484M
Cores: 8
Registry: editor.old.tab.painting=true, analyze.exceptions.on.the.fly=true, ide.allow.merge.buttons=false, debugger.watches.in.variables=false, ide.tabbedPane.dragToSplitRatio=0.00001, ide.balloon.shadow.size=0
Non-Bundled Plugins: AceJump, CMD Support, HOCON Converter, IdeaVIM, org.jetbrains.IdeaVim-EasyMotion, Key Promoter X, com.alayouni.ansiHighlight, com.github.tom-power.close-tab-left-right, GrepConsole, de.netnexus.camelcaseplugin, ideanginx9, org.jetbrains.plugins.go-template, org.jetbrains.plugins.hocon, socrates.tabshifter, com.jetbrains.lang.ejs, io.ktor.intellij.plugin, cn.wuzhizhan.plugin.mybatis, org.intellij.scala, Pythonid, org.asciidoctor.intellij.asciidoc, org.jetbrains.plugins.go, ru.adelf.idea.dotenv
Current Desktop: XFCE
```

OS version: ``Xubuntu 20.04.1``

Adding ``-XX:ErrorFile=/home/cvmocanu/temp/java_error_%p.log`` doesn't produce any log after crash (I used ``Help -> Edit custom vm options`` from IntelliJ to add this).

In case it helps, here are the other VM options (these are IntelliJ defaults; I only changed the max ``Xmx``):
```
-Xms128m
-Xmx1536m
-XX:ReservedCodeCacheSize=240m
-XX:+UseConcMarkSweepGC
-XX:SoftRefLRUPolicyMSPerMB=50
-ea
-XX:CICompilerCount=2
-Dsun.io.useCanonPrefixCache=false
-Djdk.http.auth.tunneling.disabledSchemes=""
-XX:+HeapDumpOnOutOfMemoryError
-XX:-OmitStackTraceInFastThrow
-Djdk.attach.allowAttachSelf=true
-Dkotlinx.coroutines.debug=off
-Djdk.module.illegalAccess.silent=true
-XX:MaxJavaStackTraceDepth=10000
-Dawt.useSystemAAFontSettings=lcd
-Dsun.java2d.renderer=sun.java2d.marlin.MarlinRenderingEngine
-Dsun.tools.attach.tmp.only=true
```

**Comment by ahus1** [comment]
Hello @cvmocanu - I am sorry to hear that JCEF cases you trouble. 

I have seen crashes in the past with the JavaFX preview: Sometime the crash information was only visible when starting the IDE from a command line. Please give that a try to gather more information.

Please report this crash to JetBrains; analyzing a problem within JCEF is unfortunately out-of-scope for me. 
I've seen the IDE to report a crash on the next start semi-automatically; watch out for a notification in the lower right corner.

If you can reproduce it using a JetBrains plugin that uses JCEF (for example the markdown plugin), this might help to raise the attention. 

To avoid the error, I see three alternatives that you could try: 

* upgrade your IDE - #IU-203.5419.21 is out since October 28th - it uses the same runtime AFAIK, but maybe it fixes the crash
* upgrade your JetBrains runtime using the choose-runtime-plugin to a more recent version (either a more recent release of 11.0.8, or even 11.0.9)  -> https://plugins.jetbrains.com/plugin/12836-choose-runtime
* install the JavaFX runtime for plugins and switch the JavaFX preview -> https://plugins.jetbrains.com/plugin/14250-javafx-runtime-for-plugins

As it is not clear how long JavaFX will be supported on the JetBrains platform, the third option will only buy some time; raising an issue with JetBrains will work towards a solution.

**Comment by cvmocanu** [comment]
Thanks for the quick and thorough response.

I updated to the latest EAP (XXXX.21) and the issue is gone.
The only log that I could find is ``buffer overflow detected`` printed to STDOUT (I started IntelliJ from the terminal).

For the record, it seems the following IntelliJ ticket is tracking this issue https://youtrack.jetbrains.com/issue/IDEA-251749

I think we can safely close this ticket, since it's not a problem in the Asciidoc plugin.
